src/components/Header.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useRef, useEffect } from 'react';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  title: string;
+  setTitle: React.Dispatch<React.SetStateAction<string>>;
+  handleSubmit: (event: React.FormEvent<HTMLFormElement>) => void;
+  disable: boolean;
+  todos: Todo[];
+  allCompleted: boolean;
+  handleToggleAll: () => void;
+  loading: boolean;
+};
+
+export const Header: React.FC<Props> = ({
+  title,
+  setTitle,
+  handleSubmit,
+  disable,
+  todos,
+  allCompleted,
+  handleToggleAll,
+  loading,
+}) => {
+  const focusRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (focusRef.current && !disable) {
+      focusRef.current.focus();
+    }
+  }, [disable, todos.length]);
+
+  return (
+    <header className="todoapp__header">
+      {!loading && todos.length > 0 && (
```

Comment:
In my code, this part works with only one conditional `!!todos. length`, maybe it also works for you )